src/snowflake/cli/plugins/snowpark/package_installer.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import subprocess
+import venv
+from pathlib import Path
+from tempfile import TemporaryDirectory
+
+log = logging.getLogger(__name__)
+
+
+class PackageInstaller:
+    PIP_ERROR_MESSAGE = "Running pip install for {0} {1} returned code {2}"
+    PIP_FULL_ERROR = "Full error response: {0}"
+    VENV_ERROR_MESSAGE = "Error creating venv. Full error trace: {0}"
+
+    def __init__(self, destination: Path = Path.cwd()):
```

Comment:
Let's add context manager `with venv()` that will handled proper clean up in case of any errors. Plus it will make it easier to use it  for other purposes as well.  Maybe we can do something like
```py
with venv(...) as v:
  v.pip_install(...)
```

Later we can exted this with an option to for example execute python within this venv. WDYt?